src/components/question/answers-table.cjsx

```diff
@@ -45,7 +46,31 @@ AnswersTable = React.createClass
     keySet: 'multiple-choice'
 
   getInitialState: ->
+    originalKeyScope = @getOriginalKeyScope()
+
     answer_id: null
+    originalKeyScope: originalKeyScope
+
+  componentWillReceiveProps: (nextProps) ->
+    originalKeyScope = @getOriginalKeyScope(nextProps)
+    @setState({originalKeyScope}) if originalKeyScope?
+
+    @resetToOriginalKeyScope() if not _.isNull(@props.keySet) and _.isNull(nextProps.keySet)
+
+  componentWillUnmount: ->
+    @resetToOriginalKeyScope()
+
+  getOriginalKeyScope: (props) ->
+    props ?= @props
+
+    originalKeyScope = keymaster.getScope()
+    originalKeyScope if props.keySet isnt originalKeyScope and originalKeyScope isnt @state?.originalKeyScope
+
+  resetToOriginalKeyScope: ->
```

Comment:
could be, but i think this is piece that needs it most right now.  i think eventually, we'll want to pull this out to a component or mixin but i dont know if it's worth it right now